src/main/java/com/sprint/findex_team6/repository/IndexRepository.java

```diff
@@ -40,17 +42,75 @@ public interface IndexRepository extends JpaRepository<Index, Long> {
 
   List<Index> findByIndexClassificationAndIndexNameAndFavorite(
           String indexClassification, String indexName, Boolean favorite, Pageable pageable);
-  List<Index> findByIndexClassificationCursorAsc(
-          String cursor, Pageable pageable);
+  @Query("""
+SELECT i FROM Index i
+WHERE (:indexClassification IS NULL OR i.indexClassification = :indexClassification)
+AND (:indexName IS NULL OR i.indexName LIKE %:indexName%)
+AND (:favorite IS NULL OR i.favorite = :favorite)
+AND (i.indexClassification > :cursor OR (i.indexClassification = :cursor AND i.id < :idAfter))
+""")
+  Page<Index> findByIndexClassificationCursorAsc(
+          String indexClassification,
+          String indexName,
+          Boolean favorite,
+          String cursor,
+          Long idAfter,
+          Pageable pageable
+  );
+
+
+  @Query("""
+SELECT i FROM Index i
+WHERE (:indexClassification IS NULL OR i.indexClassification = :indexClassification)
+AND (:indexName IS NULL OR i.indexName LIKE %:indexName%)
+AND (:favorite IS NULL OR i.favorite = :favorite)
+AND (i.indexClassification < :cursor OR (i.indexClassification = :cursor AND i.id < :idAfter))
+""")
+  Page<Index> findByIndexClassificationCursorDesc(
+          String indexClassification,
+          String indexName,
+          Boolean favorite,
+          String cursor,
+          Long idAfter,
+          Pageable pageable
+  );
+
+
+  @Query("""
+SELECT i FROM Index i
+WHERE (:indexClassification IS NULL OR i.indexClassification = :indexClassification)
+AND (:indexName IS NULL OR i.indexName LIKE %:indexName%)
+AND (:favorite IS NULL OR i.favorite = :favorite)
+AND (i.indexName > :cursor OR (i.indexName = :cursor AND i.id < :idAfter))
+""")
+  Page<Index> findByIndexNameCursorAsc(
+          String indexClassification,
+          String indexName,
+          Boolean favorite,
+          String cursor,
+          Long idAfter,
+          Pageable pageable
+  );
+
+
+  @Query("""
+SELECT i FROM Index i
+WHERE (:indexClassification IS NULL OR i.indexClassification = :indexClassification)
+AND (:indexName IS NULL OR i.indexName LIKE %:indexName%)
+AND (:favorite IS NULL OR i.favorite = :favorite)
+AND (i.indexName < :cursor OR (i.indexName = :cursor AND i.id < :idAfter))
+""")
+  Page<Index> findByIndexNameCursorDesc(
+          String indexClassification,
+          String indexName,
+          Boolean favorite,
+          String cursor,
+          Long idAfter,
+          Pageable pageable
+  );
 
-  List<Index> findByIndexClassificationCursorDesc(
-          String cursor, Pageable pageable);
-
-  List<Index> findByIndexNameCursorAsc(
-          String cursor, Pageable pageable);
-
-  List<Index> findByIndexNameCursorDesc(
-          String cursor, Pageable pageable);
 
   List<Index> findAll();
+
+  Page<Index> findAllByConditions(String indexClassification, String indexName, Boolean favorite, PageRequest pageRequest);
```

Comment:
이 쿼리는 작동 안할겁니다.
```suggestion

  Page<Index> findByIndexClassificationAndIndexNameAndFavorite(String indexClassification, String indexName, Boolean favorite, PageRequest pageRequest);
```
이렇게 작성하셔야 의도한 결과를 얻을 수 있어요. 근데 indexClassification이랑 indexName으로 찾으면 무조건 하나만 나옵니다. 두 개가 유니크 제약 조건으로 묶여 있어요. 그래서 쿼리를 다시 작성하셔야 합니다.
```java
  @Query("""
SELECT i FROM Index i
WHERE (:indexClassification IS NULL OR i.indexClassification = :indexClassification)
AND (:indexName IS NULL OR i.indexName LIKE %:indexName%)
AND (:favorite IS NULL OR i.favorite = :favorite)
ORDER BY i.id ASC
""")
```
이런 느낌으로 쿼리를 날리면 될거같은데, 제가 틀렸을 수도 있으니 한번 생각해보시고 수정해주세요.